autopush-common/src/reliability.rs

```diff
@@ -363,16 +382,18 @@ impl PushReliability {
                 // it should return `nil`, in which case we record a soft error.
                 // This could also be strung together as a cascade of functions, but it's broken
                 // out to discrete steps for readability.
+                if result == redis::Value::Nil {
+                    warn!("🔍⚠🪈 {id} - Pipe failed, retry.");
+                    return Ok(None);
+                }
                 if let Some(operations) = result.as_sequence() {
                     // We have responses, the first items report the state of the commands,
                     // the final line is a list of command results.
                     if let Some(result_values) = operations.last() {
                         if let Some(results) = result_values.as_sequence() {
                             // The last command should contain the prior state. If it returned `Nil`
                             // for some, unexpected reason, note the error.
-                            if new != ReliabilityState::Received
-                                && Some(&redis::Value::Nil) == results.last()
-                            {
+                            if Some(&redis::Value::Nil) == results.last() {
```

Comment:
I meant here: the comment suggests we're checking the prior state from the `SET/GET`, but we're not when `== Received` -- we're instead of checking the final call of an `HINCR`. Per the docs I guess `HINCR` never fails -- always returning the new value -- if so it wouldn't trigger this, but we could clarify this check either by adjusting the comment or keeping the previous code that wraps the check in a `new != ReliabilityState::Received` also clarifies it and the comment as is

Reply:
Ah! I see what you mean. I'll rework the comment to be more correct and general.